### DISCUSSION TITLE
No Information available from Inverter

### DISCUSSION THREAD

**Discussion by MaxDaNr1** [tbnobody/OpenDTU#307] [Q&A]
Hallo zusammen,

ich bin ganz neu hier auf github und weil ich vom OpenDTU auf Twitter gelesen hatte. OpenDTU ist mein allererstes Projekt.
Ich bin also voll der Noob^^
Die Anleitung war aber so gut, dass ich es mir zugetraut habe. Ich habe aber schon lange Bock auf derartige Projekte gehabt, deswegen musste ich jetzt mal endlich anfangen.

Der Aufbau steht also soweit, ist mit meinem WLAN erfolgreich verbunden und ich konnte alle Settings soweit einstellen. Leider bekomme ich nur keine Verbindung zu meinem HM-1500. Er ist definitiv am Laufen, da ich ihn parallel mit einem Shelly Plug S überwache. 

Ich bekomme auf der Oberfläche von OpenDTU folgende Meldung:
"No Information available
Did not receive any valid data from the inverter till now. Still trying..."

Distanz zu HM-1500: circa drei Meter direkte Sichtverbindung.

Unter: Info -> System steht:
Chip Status: connected
Chip Type: nRF24L01+

Wie kann ich herausfinden was falsch gelaufen ist? Ich muss zugeben, dass ich am nRF24L01+ zum allerersten Mal in meinem Leben gelötet habe und vielleicht habe ich hier ja etwas falsch gemacht, aber es wird ja angegeben, dass der nRF24L01+ connected ist. Was könnte noch falsch gelaufen sein.

Ich hoffe jemand hat die Geduld einem Noob zu helfen!
Danke im Voraus

**Comment by MaxDaNr1** [comment]
Folgendes habe mich mit Putty serial ausgelesen:

_Nothing received, resend count exeeded
TX AlarmData Channel: 76 --> 15 74 40 40 71 80 17 13 04 80 11 00 63 66 C1 09 00 00 00 00 00 00 00 00 B4 E0 98
RX Period End
All missing
Nothing received, resend whole request
TX AlarmData Channel: 76 --> 15 74 40 40 71 80 17 13 04 80 11 00 63 66 C1 09 00 00 00 00 00 00 00 00 B4 E0 98
RX Period End
All missing
Nothing received, resend whole request
TX AlarmData Channel: 76 --> 15 74 40 40 71 80 17 13 04 80 11 00 63 66 C1 09 00 00 00 00 00 00 00 00 B4 E0 98
RX Period End
All missing
Nothing received, resend whole request
TX AlarmData Channel: 76 --> 15 74 40 40 71 80 17 13 04 80 11 00 63 66 C1 09 00 00 00 00 00 00 00 00 B4 E0 98
RX Period End
All missing
Nothing received, resend whole request
TX AlarmData Channel: 76 --> 15 74 40 40 71 80 17 13 04 80 11 00 63 66 C1 09 00 00 00 00 00 00 00 00 B4 E0 98
RX Period End
All missing
Nothing received, resend count exeeded
TX SystemConfigPara Channel: 76 --> 15 74 40 40 71 80 17 13 04 80 05 00 63 66 C1 09 00 00 00 00 00 00 00 00 A0 F4 8C
RX Period End
All missing
Nothing received, resend whole request
TX SystemConfigPara Channel: 76 --> 15 74 40 40 71 80 17 13 04 80 05 00 63 66 C1 09 00 00 00 00 00 00 00 00 A0 F4 8C
RX Period End
All missing
Nothing received, resend whole request
TX SystemConfigPara Channel: 76 --> 15 74 40 40 71 80 17 13 04 80 05 00 63 66 C1 09 00 00 00 00 00 00 00 00 A0 F4 8C
RX Period End
All missing
Nothing received, resend whole request
TX SystemConfigPara Channel: 76 --> 15 74 40 40 71 80 17 13 04 80 05 00 63 66 C1 09 00 00 00 00 00 00 00 00 A0 F4 8C
RX Period End
All missing
Nothing received, resend whole request
TX SystemConfigPara Channel: 76 --> 15 74 40 40 71 80 17 13 04 80 05 00 63 66 C1 09 00 00 00 00 00 00 00 00 A0 F4 8C
RX Period End
All missing
Nothing received, resend count exeeded_

**Comment by stefan123t** [comment]
Er scheint zu senden aber für die Antwort kommt kein Interrupt. Also entweder IRQ Pins ändern oder das NRF24 Modul tauschen, wenn Du schon selber meinst Du hast da was falsch gelötet ?

**Reply by MaxDaNr1** [reply]
Ich habe mir mittlerweile ein neues NRF24 Modul besorgt bei dem ich nicht selbst löten musste. Das Problem besteht weiterhin. Unter SystemInfo -> Radio Information heisst es auch, dass das Modul verbunden und erkannt wird. Aber es kommen keine Infos vom WR. Die ID vom WR habe ich auhc schon zig Male überprüft. Was könnte ich denn am IRG Pin ändern?

**Comment by stefan123t** [comment]
Also wir empfehlen stark einen 10-100uF Elektrolyt Kondensator an GND und 3.3V am NRF Modul anzulöten damit die Strom/Spannungsversorgung beim Senden nicht einbricht. Polung beachten sonst Rumms oder kaputt !

**Comment by scoace** [comment]
Hi, hast du mal AhoyDTU getestet. Habe hier auch schonmal 2h mit OpnDTU investiert und nichts ging. Selbiges mit AhoyDTU gleiches Modul und 2min später Daten vom Wechselrichter.

**Reply by MaxDaNr1** [reply]
Ja, ich habe derzeit auch Ahoy installiert. Hier mal die Fehler aus der Console:

12:43:17 I: (#0) sendTimePacket
12:43:17 I: sendTimePacket 5
12:43:17 I: TX 27B Ch75 | 15 74 40 40 71 82 07 13 04 80 05 00 63 ec c5 52 00 00 00 00 00 00 00 00 5c b2 f1
12:43:19 W: while retrieving data: last frame missing: Request Retransmit
12:43:19 I: (#0) sendTimePacket
12:43:19 I: sendTimePacket 5
12:43:19 I: TX 27B Ch3 | 15 74 40 40 71 82 07 13 04 80 05 00 63 ec c5 52 00 00 00 00 00 00 00 00 5c b2 f1
12:43:21 W: while retrieving data: last frame missing: Request Retransmit
12:43:21 I: (#0) sendTimePacket
12:43:21 I: sendTimePacket 5
12:43:21 I: TX 27B Ch23 | 15 74 40 40 71 82 07 13 04 80 05 00 63 ec c5 52 00 00 00 00 00 00 00 00 5c b2 f1
12:43:24 W: while retrieving data: last frame missing: Request Retransmit
12:43:24 I: (#0) sendTimePacket
12:43:24 I: sendTimePacket 5
12:43:24 I: TX 27B Ch40 | 15 74 40 40 71 82 07 13 04 80 05 00 63 ec c5 52 00 00 00 00 00 00 00 00 5c b2 f1

**Reply by scoace** [reply]
Läuft Ahoy bei dir?

**Reply by MaxDaNr1** [reply]
Nein, leider werden dort auch keine Daten vom WR empfangen. Er sendet fleißig (Tx) aber es wird nichts empfangen. Komischerweise steht in der Console zwar "while retrieving data: last frame missing: Request Retransmit" und man könnte meinen, dass er Bruchstücke empfängt. Wenn ich aber eine beliebige Seriennummer angebe spuckt er das gleiche in der Console aus.

**Comment by thb584Git** [comment]
Hatte seit heute das gleiche Problem nachdem ich meinen Aufbau auf eine Lochrasterplatine verpflanzt habe. 
Davor lief ein ESP32 Node MCU als fligender Test-Aufbau mit kurzen Jumper Kabeln zur NFC24 versorgt über USB, wahlweise aus PC-Port dann über Steckerladegerät. Lief eine Woche ohne dass Übertragungsfehler aufgefallen wären.
Neuer Aufbau: DC/DC von 5V-18V auf 3,3V geblockt mit Elkos und Keramik-C, ESP32 und NRF24 nun beide von 3,3V versorgt mit dem Resultat "All mssing" keine Daten
Nach längerem Suchen läufts nun auch damit. Bei mir waren es zwei Probleme:
1. Die Versorgungsspannung ist äußerst kritisch. Bei mir steckt das NRF24 in einem Sockel und die C's sind auf der Lochrasterplatine noch 1-2cm entfernt. DIe Verbindungsdrähte der Versorgungsspannung sind 0,4-0,6er DArht,  also gut und trotzdem hatte ich erst mit einem low ESR-Elko direkt auf dem NRF24 Modul die ersten Daten empfangen aber immer noch jede Menge Widerholungen.
2. Der Abstand der Antennen. Obwohl ich die beiden boards mit den Antenne nach Außen montiert hatte scheinen sich beide zu stören. Nachdem ich das RF-Modul mit Jmper Kabel abgesetzt habe läuft das Interface wieder.  Die Sendeleistung hat bei mir nicht wirklich Einfluss. Es gibt immer weider mal retransmits aber die Daten kommen bei jeder Abfrage zurück.
3. Elko testweise abgelötet - kein Datenempfang bei gleichem Antennenabstand

Also bei mit läuft es nur mit Elko direkt auf NRF24 (ist eins von ALiEx, vielleicht gits da Unterschiede) und Antenne mit mindestens 10cm Abstand.
Viel Erfolg

**Reply by MaxDaNr1** [reply]
Bei mir funktioniert es jetzt ohne Elko einwandfrei. Es lag bei mir am nRF-Modul (siehe unten).

**Comment by MaxDaNr1** [comment]
So, ich hatte am Wochenende ein wenig Zeit und konnte mal was ausprobieren. Jetzt hat es endlich geklappt und bei mir läuft es nach 3 Monaten endlich. Bei mir war das Problem das nRF+ Modul.

Ich habe mir im Internet ein komplett fertiges OpenDTU mit Gehäuse bestellt. Nur so konnte ich mit wenig finanziellen und zeitlichen Aufwand auf Fehlersuche gehen. Also das bestellte OpenDTU angeschlossen, alles eingerichtet und ratz-fatz hatte ich Daten vom WR. So konnte ich schonmal ausschließen, dass es am WR liegt. Dann habe ich das bestellte OpenDTU auseinander gebastelt und nach und nach die Komponenten von meinem selbst gebauten OpenDTU ausgetauscht. Ich habe sogar die Jumper-Kabel getauscht, um auch diese ausschließen zu können. Letzendlich lag es bei mir definitiv am nRF+Modul. Mein selbst gebauten OpenDTU funktioniert mit dem nRF+Modul vom fertigen OpenDTU einwandfrei. Auch Netzteil und USB-Kabel kann ich ausschließen.

Beide nRF+Module haben einem Punkt auf dem Chip (und kein Quadrat) und auf beiden ist ein Plus. Lediglich einzelne Komponenten unterscheiden sich.
![nRF- Module im Vergleich](https://user-images.githubusercontent.com/117392361/219951946-f396c47f-0609-4b05-a418-7e1e662e5d78.jpg)

Links das NICHT funktionierende Modul und rechts das funktionierende Modul. Leider bekommt meine Handykamera es nicht hin einen besseren Fokus einzustellen.
Da ich vorher bereits schon einmal ein anderes nRF-Modul probiert hatte, hatte ich eigentlich ausgeschlossen, dass es am nRF-Modul liegt, aber scheinbar kann man im Internet echt viele falsche Module bestellen. Ich hoffe das hilft vielleicht weiter.

**Reply by thb584Git** [reply]
Freut mich für Dich dass es läuft. Das mit den fakes glaube ich allerdings eher nicht. Der Baustein ist single source, das ist nicht bei teuren OP-Amps wo man einfach einen billigen Typen anders stempelt oder Bauteile wie Transistoren die weniger können hernimmt. Meine Module haben bei AliEx um die 50 cent gekostet und laufen. 
Ich denke das design auf dem board ist kritisch bzgl. Masse. Deine beiden haben unterschiedliche Silk-screens (Aufdrucke). Eins mit abgerundeten Ecken. Somit dürfe das layout sich unterscheiden. Check mal die Durchkontaktierungen (kleine Löcher) ob das funktionirenede mehr davon hat. Außerdem scheinen die BE rechts kleiner, was bei den Kond. möglicherweise weniger Kapazität bedeutet. Ich würde vermuten das schlechte läuft auch mit zusätzlichem Kondensator. Bei mir habe ich nun einen 10uF SMD direkt am Stecker, das hat allerdings auch eckige Aufrucke. Wo stammt dein schlechtes Modul her?
![NRF24](https://user-images.githubusercontent.com/54303509/219954213-53b9e3f3-a6da-4c11-b574-dc3e924c7a49.jpg)

**Reply by Brain-Kiel** [reply]
Nachdem ich ebenfalls ein Problem mit den NRFs hatte und ich mir das Elko Löten sparen wollte, habe ein 5V Breakboard (mit Step-Down) zwischengeschaltet, welches direkt an der Vcc hängt.
![grafik](https://user-images.githubusercontent.com/110300565/219956290-ce5e08a3-824f-4c5f-a29e-7cbeb427cebd.png)
Das Teil hat eine Betriebs-LED. Wenn die Spannung zusammenbricht, sieht man es flackern. Das hatte ich bei einem ESP Clone...

**Reply by MaxDaNr1** [reply]
Leider kann ich technisch nicht mehr ganz folgen. Daher habe ich mir mal die Mühe gemacht und bessere Fotos gemacht. Hier die Nahaufnahmen:
![nRF- Module Close-Up](https://user-images.githubusercontent.com/117392361/220186275-0f9c6bfc-7873-49f0-bc9f-8b40ba6b0431.jpg)
Ich hatte das nicht funktionsfähige Modul auf Ebay beim Makershop gekauft.

**Reply by thb584Git** [reply]
Gut dass Du ein anders erhalten hast das funktioniert. Soweit ich das auf dem Foto sehen kann scheinen auch bei dem nicht funktionierenen die Durchkontaktierungen zahlreich vorhanden. Kann an vielem liegen: andere Kondensatoren, Exemplarfehler des IC, Lötprobem, layout, Leiterplatte. Ist halt aufwändig das rauszufinden und nicht wirtschaftlich. Deshalb wenn du einen Kondenstor anlöten kannst wäre es einen Versuch wert, ansonsten in den Elektroschrott.
Ärgerlich wenn das auch aus einer eigentlich seriösen Quelle passiert. Aber die kaufen halt auch nur in China und haben keine Qualitätssicherung außer vielleicht wenn die Kunden sich gehäuft beschweren.
Viel Spass mit Deinem OpenDTU
